Seymour/Matrix/TotalUnimodularityTest.lean

```diff
@@ -48,13 +48,29 @@ theorem Matrix.testTotallyUnimodular_eq_isTotallyUnimodular {m n : ℕ} (A : Mat
 instance {m n : ℕ} (A : Matrix (Fin m) (Fin n) ℚ) : Decidable A.IsTotallyUnimodular :=
   decidable_of_iff _ A.testTotallyUnimodular_eq_isTotallyUnimodular
 
+def Matrix.square_set_submatrix {m n : ℕ} (A : Matrix (Fin m) (Fin n) ℚ)
+      (r : Finset (Fin m)) (c : Finset (Fin n)) (h : #r = #c) :=
+  @Matrix.submatrix (Fin (#r)) _ _ (Fin (#r)) ℚ A ((r.sort (· ≤ ·))[·]) (fun p ↦ (c.sort (· ≤ ·))[p]'(by
+      rw [Finset.length_sort, show c.card = r.card by simp_all only [Fintype.card_coe]]
+      calc
+        p.val < #{ x // x ∈ r } := Fin.isLt p
+        _ = r.card := by simp_all only [Fintype.card_coe]
+      ))
 
 /-- Faster algorithm for testing total unimodularity but without formal guarantees. -/
```

Comment:
Is there a reason to remove it prior to finishing the fastest version?

Reply:
I can keep it - I marked this PR as a draft in hopes that it would be finished by the time it would be merged, but I can re-add it if you merge it before then :+1:

Reply:
All right, I will wait before merging!